src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -692,7 +687,9 @@ JRT_LEAF(BasicType, Deoptimization::unpack_frames(JavaThread* thread, int exec_m
   // It is actually ok to allocate handles in a leaf method. It causes no safepoints,
   // but makes the entry a little slower. There is however a little dance we have to
   // do in debug mode to get around the NoHandleMark code in the JRT_LEAF macro
-  ResetNoHandleMark rnhm; // No-op in release/product versions
+  // Beware though because allocating Handles must have a HandleMark or else the
+  // Handles will be leaked.
+  ResetNoHandleMark rnhm;
```

Comment:
So I'm confused. The bug description says:

> the ResetNoHandleMark is unneeded

but it is still here.

Reply:
This one is needed because it's in a JRT_LEAF.  The one above in the JRT_BLOCK_ENTRY is unneeded (along with the comment that matched the one in this unpack_frames() function.)

Reply:
Correction, I didn't remove a ResetNoHandleMark in fetch_unroll_info, because there wasn't one there.  It was removed already.  I just removed the comment that described the code that isn't there anymore.